File: ios/Sources/CollectDataPluginPlugin/CollectDataPluginPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN macro.
CAP_PLUGIN(CollectDataPluginPlugin, "CollectDataPlugin",
    CAP_PLUGIN_METHOD(getPlatformVersion, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(fetchSMSInfo, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(fetchDeviceData, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(syncSMSData, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(syncDeviceLocationAppsData, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(syncAllData, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(startBackGroundSyncProcess, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(stopBackGroundSyncProcess, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(disableSyncs, CAPPluginReturnPromise);
) 

File: ios/Sources/CollectDataPluginPlugin/CollectDataPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CollectDataPluginPlugin)
public class CollectDataPluginPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CollectDataPluginPlugin"
    public let jsName = "CollectDataPlugin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = CollectDataPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
